File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        graph = {}
        for index, value in enumerate(nums):
            complement = target - value
            if complement in graph.keys():
                return[graph[complement], index]
            graph[value] = index
            
            

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let l=0, r = numbers.length - 1;
        while(l<r){
            let k = numbers[r] + numbers[l]
            if (k < target){
                l++
            }else if(k > target){
                r--
            }
            else{
                return [l+1,r+1]
            }
        }
    }
}
